pkg/storage/backend-storage/backend-storage.go

```diff
@@ -0,0 +1,101 @@
+package backendstorage
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/utils/pointer"
+
+	virtconfig "kubevirt.io/kubevirt/pkg/virt-config"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	corev1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/kubecli"
+)
+
+const (
+	PVCPrefix = "persistent-state-for-"
+	PVCSize   = "10Mi"
+)
+
+func PersistentPVCForVMI(vmi *corev1.VirtualMachineInstance) string {
+	return PVCPrefix + vmi.Name
+}
+
+func HasPersistentTPMDevice(vmi *corev1.VirtualMachineInstance) bool {
+	if vmi.Spec.Domain.Devices.TPM != nil &&
+		vmi.Spec.Domain.Devices.TPM.Persistent != nil &&
+		*vmi.Spec.Domain.Devices.TPM.Persistent {
+		return true
+	}
+
+	return false
+}
+
+func isBackendStorageNeeded(vmi *corev1.VirtualMachineInstance) bool {
+	return HasPersistentTPMDevice(vmi)
+}
+
+func CreateIfNeeded(vmi *corev1.VirtualMachineInstance, clusterConfig *virtconfig.ClusterConfig, client kubecli.KubevirtClient) error {
+	if !isBackendStorageNeeded(vmi) {
+		return nil
+	}
+
+	_, err := client.CoreV1().PersistentVolumeClaims(vmi.Namespace).Get(context.Background(), PVCPrefix+vmi.Name, metav1.GetOptions{})
+	if err == nil {
+		return nil
+	}
+	if !errors.IsNotFound(err) {
+		return err
+	}
+
+	modeFile := v1.PersistentVolumeFilesystem
+	storageClass := clusterConfig.GetVMStateStorageClass()
+	if storageClass == "" {
+		return fmt.Errorf("backend VM storage requires a backend storage class defined in the custom resource")
+	}
+	ownerReferences := vmi.OwnerReferences
+	if len(vmi.OwnerReferences) == 0 {
+		// If the VMI has no owner, then it did not originate from a VM.
+		// In that case, we tie the PVC to the VMI, rendering it quite useless since it wont actually persist.
+		// The alternative is to remove this `if` block, allowing the PVC to persist after the VMI is deleted.
+		// However, that would pose security and littering concerns.
+		ownerReferences = []metav1.OwnerReference{{
+			APIVersion:         corev1.VirtualMachineInstanceGroupVersionKind.GroupVersion().String(),
+			Kind:               corev1.VirtualMachineInstanceGroupVersionKind.Kind,
+			Name:               vmi.Name,
+			UID:                vmi.UID,
+			Controller:         pointer.BoolPtr(true),
+			BlockOwnerDeletion: pointer.BoolPtr(true),
+		}}
```

Comment:
Probably this can be simplified with `NewControllerRef` from `k8s.io/apimachinery/pkg/apis/meta/v1`:

```suggestion
			*metav1.NewControllerRef(vmi, v1.VirtualMachineInstanceGroupVersionKind),
		}}
```